drivers/flash/flash_stm32h7x.c

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2020 Vossloh Cogifer
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <drivers/flash.h>
+#include <init.h>
+#include <soc.h>
+
+#include "flash_stm32.h"
+
+#define LOG_DOMAIN flash_stm32h7
+#define LOG_LEVEL CONFIG_FLASH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+#define STM32H7X_SECTOR_SHIFT 17
```

Comment:
Some H7 references use 8K sectors.

Reply:
```diff
-#define STM32H7X_SECTOR_SHIFT 17
+
+#define LOG2CEIL(x) ((((x) <= 2048) ? 11 : (((x) <= 4096)?12:(((x) <= 8192) ? \
+       13 : (((x) <= 16384) ? 14 : (((x) <= 32768) ? 15:(((x) <= 65536) ? \
+       16 : (((x) <= 131072) ? 17 : (((x) <= 262144) ? 18:(((x) <= 524288) ? \
+       19 : (((x) <= 1048576) ? 20 : (((x) <= 2097152) ? \
+       21 : (((x) <= 4194304) ? 22 : (((x) <= 8388608) ? \
+       23 : (((x) <= 16777216) ? 24 : (((x) <= 33554432) ? \
+       25 : (((x) <= 67108864) ? 26 : (((x) <= 134217728) ? \
+       27 : (((x) <= 268435456) ? 28 : (((x) <= 536870912) ? \
+       29 : (((x) <= 1073741824) ? 30 : (((x) <= 2147483648) ? \
+       31 : 32))))))))))))))))))))))
+#define STM32H7X_SECTOR_SHIFT LOG2CEIL(FLASH_BANK_SIZE / FLASH_SECTOR_TOTAL)
```
What do you think about this ? I copied the macro LOG2CEIL from `include/linker/linker-tool-mwdt.h`
This way the SECTOR_SHIFT is computed automatically for every devices in the family.

Reply:
What about simplifying a bit things and use the following in `get_sector` function:
```
static unsigned int get_sector(off_t offset)
{
	return offset / FLASH_SECTOR_SIZE;
}
```